lewis/adapters/modbus.py

```diff
@@ -307,7 +308,9 @@ def process(self, data):
                 'Request: %s', str(['{:#04x}'.format(c) for c in request.to_bytearray()]))
 
             handler = self._get_handler(request.fcode)
-            response = handler(request)
+
+            with device_lock:
+                response = handler(request)
```

Comment:
Actually, thinking about this one, it's probably more similar to TCP stream than to EPICS and it's fine either way. Still, just to avoid locking/unlocking many times I would put the lock outside of the loop.